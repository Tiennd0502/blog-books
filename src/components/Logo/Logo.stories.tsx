import React from 'react';
import Storybook from '@storybook/react';

import IMAGES from '@constants/images';
import Logo from './index';

export default {
  title: 'Components/Logo',
  component: Logo,
} as Storybook.ComponentMeta<typeof Logo>;

const Template: Storybook.ComponentStory<typeof Logo> = (args) => (
  <Logo {...args} />
);

export const Default = Template.bind({});
Default.args = {
  url: IMAGES.LOGO_URL,
  alt: IMAGES.LOGO_ALT,
};
