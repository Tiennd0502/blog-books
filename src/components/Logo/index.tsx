import React from 'react';
import Image from 'next/image';
import Link from 'next/link';
import { Box } from '@chakra-ui/react';

import IMAGES from '@constants/images';
import imageLoader from '@helpers/image';

const Logo: React.FC = () => {
  return (
    <Link href="/">
      <Box cursor="pointer">
        <Image
          src={IMAGES.LOGO_URL}
          alt={IMAGES.LOGO_ALT}
          layout="responsive"
          width="852"
          height="213"
          placeholder="blur"
          blurDataURL={`/_next/image?url=${IMAGES.LOGO_URL}&w=16&q=1`}
          loader={imageLoader}
          priority
          sizes="50vw"
        />
      </Box>
    </Link>
  );
};

export default Logo;
