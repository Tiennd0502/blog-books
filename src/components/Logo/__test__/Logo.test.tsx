import { render } from '@testing-library/react';

import IMAGES from '@constants/images';
import Logo from '../index';

describe('Logo component', () => {
  test('Should render Logo component', () => {
    const { getByAltText } = render(<Logo />);
    const image = getByAltText(IMAGES.LOGO_ALT) as HTMLImageElement;

    expect(image).toBeInTheDocument();
  });

  test('Matches snapshot', () => {
    const { container } = render(<Logo />);

    expect(container).toMatchSnapshot();
  });
});
