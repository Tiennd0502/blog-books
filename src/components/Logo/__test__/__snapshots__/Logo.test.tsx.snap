// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component Matches snapshot 1`] = `
<div>
  <div
    class="css-4g6ai3"
  >
    <span
      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 25% 0px 0px 0px;"
      />
      <img
        alt="49th Shelf"
        data-nimg="responsive"
        decoding="async"
        sizes="100vw"
        src="/_next/image?url=/images/logo.png&w=3840&q=75"
        srcset="/_next/image?url=/images/logo.png&w=640&q=75 640w, /_next/image?url=/images/logo.png&w=750&q=75 750w, /_next/image?url=/images/logo.png&w=828&q=75 828w, /_next/image?url=/images/logo.png&w=1080&q=75 1080w, /_next/image?url=/images/logo.png&w=1200&q=75 1200w, /_next/image?url=/images/logo.png&w=1920&q=75 1920w, /_next/image?url=/images/logo.png&w=2048&q=75 2048w, /_next/image?url=/images/logo.png&w=3840&q=75 3840w"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/_next/image?url=/images/logo.png&w=16&q=1);"
      />
      <noscript />
    </span>
  </div>
</div>
`;
